app/models/user.rb

```diff
@@ -5,7 +5,9 @@ class User < ApplicationRecord
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :trackable, :validatable
 
-  validates_presence_of :first_name, :last_name
+  validates_presence_of :first_name, :last_name, on: :create
+  validates_presence_of :first_name, :last_name, on: :update, unless: Proc.new {encrypted_password != encrypted_password_was}
```

Comment:
This is an unlikely situation, but this fix, as is, allows a user to change their password and - at the same time - set their first and/or last name to nil.

I think this might fix this and not introduce any other problems:
```
validates_presence_of :first_name, :last_name, on: :update,
    unless: Proc.new { encrypted_password != encrypted_password_was &&
                       !(first_name_changed? || last_name_changed?) }
```
I did some limited testing locally and this prevented the situation I mention.

But, can you see any downside to this?

Reply:
No downside, and I think your approach is cleaner than mine. Instead of focussing on the password problem (which is only a symptom), it is better to realise we only have to validate the presence of first- and lastname if either change (or in case of a new record).